Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -685,10 +698,41 @@ bool Renderer::IsFrameDumping()
   if (SConfig::GetInstance().m_DumpFrames)
     return true;
 
-  ShutdownFrameDumping();
   return false;
 }
 
+void Renderer::DoDumpFrame()
+{
+  UpdateFrameDumpTexture();
+
+  auto result = m_dump_texture->Map();
+  if (result.has_value())
+  {
+    auto raw_data = result.value();
+    DumpFrameData(raw_data.data, raw_data.width, raw_data.height, raw_data.stride,
+                  AVIDump::FetchState(m_last_xfb_ticks));
+  }
+}
+
+void Renderer::UpdateFrameDumpTexture()
+{
+  int target_width, target_height;
+  std::tie(target_width, target_height) = CalculateOutputDimensions(
+      m_last_xfb_texture->GetConfig().width, m_last_xfb_texture->GetConfig().height);
+  if (m_dump_texture == nullptr ||
+      m_dump_texture->GetConfig().width != static_cast<u32>(target_width) ||
+      m_dump_texture->GetConfig().height != static_cast<u32>(target_height))
+  {
+    TextureConfig config;
+    config.width = target_width;
+    config.height = target_height;
+    config.rendertarget = true;
+    m_dump_texture = g_texture_cache->CreateTexture(config);
+  }
+  m_dump_texture->CopyRectangleFromTexture(m_last_xfb_texture, m_last_xfb_region,
```

Comment:
Since you're using alpha blending for the final "scanout", what values get written to the frame dump? In the past we've had issues with non-opaque alpha being encoded, but it depends on the codec.

Reply:
Hmm, good question, I believe it ends up black but I will test this.